### PR TITLE
feat: Added Parameter Descriptions for Plugins

### PR DESCRIPTION
**Pull request by dmikolay**
### Proposed Change
- Added descriptions to each parameters within the plugins
- Changed two instances of a location parameter into timezone type

##### Checklist
- [ ] Changes are tested
- [ ] CI has passed

### CODE REVIEW

File: plugins/syslog_logs.yaml [hunk 1]
```diff
@@ -40,9 +49,11 @@ parameters:
     default: "1.2"
   # Max message size for TCP only
   - name: max_log_size
+    description: Maximum amount of bytes taken up by a log file
```
Comment (BinaryFissionGames) [inline plugins/syslog_logs.yaml:52]:
```suggestion
    description: Maximum number of bytes for a single TCP message. Only applicable when connection_type is TCP 
```

File: plugins/vcenter_logs.yaml [hunk 2]
```diff
@@ -3,22 +3,27 @@ title: VMware vCenter
...
     type: int
     default: 5140
   - name: listen_ip
+    description: A vCenter ip address of the form `<ip>`
     type: string
     default: "0.0.0.0"
   - name: max_buffer_size
     type: string
     default: "1024kib"
     required: false
```
Comment (BinaryFissionGames) [inline plugins/vcenter_logs.yaml:16]: Looks like this one got missed

**Review by BinaryFissionGames — COMMENTED** [review]
Just took a cursory look, this looks great in general.

File: plugins/ingress_nginx_logs.yaml [hunk 3]
```diff
@@ -3,17 +3,20 @@ title: Ingress NGINX
 description: Log parser for Ingress NGINX
 parameters:
   - name: file_path
+    description: description: Specify a single path or multiple paths to read one or many files. You may also use a wildcard (*) to read multiple files within a directory
```
Comment (BinaryFissionGames) [inline plugins/ingress_nginx_logs.yaml:6]:
```suggestion
    description: Specify a single path or multiple paths to read one or many files. You may also use a wildcard (*) to read multiple files within a directory
```

File: plugins/vcenter_logs.yaml (hunk 2 above)
Comment (dmikolay) [reply plugins/vcenter_logs.yaml:16]: Oops, I went in and fixed it

File: plugins/ingress_nginx_logs.yaml (hunk 3 above)
Comment (dmikolay) [reply plugins/ingress_nginx_logs.yaml:6]: Good catch

File: plugins/cisco_asa_logs.yaml [hunk 4]
```diff
@@ -3,9 +3,11 @@ title: Cisco ASA
 description: Log parser for Cisco ASA
 parameters:
   - name: listen_port
+    description: A port which the agent will listen for syslog messages
     type: int
     default: 5140
   - name: listen_ip
+    description: A syslog ip address of the form `<ip>`
```
Comment (cpheps) [inline plugins/cisco_asa_logs.yaml:10]: I think we can drop the `of the form <ip>` part? Not sure it adds much help to the user. Probably the same in every plugin that has this.

File: plugins/iis_logs.yaml [hunk 5]
```diff
@@ -3,29 +3,36 @@ title: IIS
...
-    default: "UTC"
+  - name: timezone
```
Comment (cpheps) [inline plugins/iis_logs.yaml:14]: Can we bump the version of this plugin to `0.1.0` since we modified this type?

File: plugins/syslog_logs.yaml [hunk 6]
```diff
@@ -3,34 +3,43 @@ title: Syslog
...
       - rfc5424
     default: rfc5424
-  - name: location
-    type: string
+  - name: timezone
+    description: Timezone to use when parsing the timestamp
+    type: timezone
     default: UTC
   - name: listen_address
+    description: Parameter Deprecated Use 'listen_ip' and 'listen_port' instead
```
Comment (cpheps) [inline plugins/syslog_logs.yaml:24]: I don't see a `listen_ip` and `listen_port` in this plugin. We likely should change this over to that format at some point but right now this description is asking the user to use parameters that don't exist.

File: plugins/cisco_asa_logs.yaml (hunk 4 above)
Comment (dmikolay) [reply plugins/cisco_asa_logs.yaml]: Sure, went through and took that part out

File: plugins/cisco_meraki_logs.yaml [hunk 7]
```diff
@@ -3,9 +3,11 @@ title: Cisco Meraki
 description: Log parser for Cisco Meraki
 parameters:
   - name: listen_port
+    description: A port which the agent will listen for syslog messages
     type: int
     default: 5140
   - name: listen_ip
+    description: A syslog ip address
```
Comment (BinaryFissionGames) [inline plugins/cisco_meraki_logs.yaml:10]: This makes it sound like this is an IP address of syslog or something; This is actually the local IP to bind to (syslog is the client, we're the server)
```suggestion
    description: The local IP address to listen for syslog connections on
```

File: plugins/couchbase_logs.yaml [hunk 8]
```diff
@@ -3,55 +3,70 @@ title: Couchbase
...
+    description: The absolute path to the Couchbase babysitter logs
     type: "[]string"
     default: 
       - "/opt/couchbase/var/lib/couchbase/logs/babysitter.log"
   - name: enable_xdcr_log
+    description: Enable to collect Couchbase Cross Datacenter Replication logs
     type: bool
     default: false
   - name: xdcr_log_path
+    description: The absolute path to the Couchbase cross datacenter replication logs
```
Comment (BinaryFissionGames) [inline plugins/couchbase_logs.yaml:64]: Just for consistencies sake:
```suggestion
    description: The absolute path to the Couchbase Cross Datacenter Replication logs
```

File: plugins/csv_logs.yaml [hunk 9]
```diff
@@ -3,21 +3,26 @@ title: CSV
...
     supported:
       - beginning
       - end
     default: end
   - name: exclude_log_paths
+    description: Specify a single path or multiple paths to exclude one or many files from being read. You may also use a wildcard (*) to exclude multiple files from being read within a directory.
     type: "[]string"
     default: []
   - name: header
+    description: Comma Separated header string to be used as keys
```
Comment (BinaryFissionGames) [inline plugins/csv_logs.yaml:21]:
```suggestion
    description: Comma separated header string to be used as keys
```

**Review by cpheps — APPROVED** [review]

File: plugins/esxi_logs.yaml [hunk 10]
```diff
@@ -3,18 +3,23 @@ title: VMware ESXi
 description: Log parser for VMware ESXi
 parameters:
   - name: listen_port
+    description: A port which the agent will listen for ESXi messages
     type: int
     default: 5140
   - name: listen_ip
+    description: A ESXi ip address
```
Comment (BinaryFissionGames) [inline plugins/esxi_logs.yaml:10]:
```suggestion
    description: The local IP address to listen for ESXi connections on
```

File: plugins/tcp_logs.yaml [hunk 11]
```diff
@@ -3,25 +3,33 @@ title: TCP
 description: Log receiver for TCP
 parameters:
   - name: listen_port
+    description: A port which the agent will listen for tcp messages
     type: int
     required: true
   - name: listen_ip
+    description: An ip address
```
Comment (BinaryFissionGames) [inline plugins/tcp_logs.yaml:10]:
```suggestion
    description: The local IP address to listen for TCP connections on
```

File: plugins/ubiquiti_logs.yaml [hunk 12]
```diff
@@ -3,12 +3,15 @@ title: Ubiquiti
 description: Log parser for Ubiquiti Unifi Devices
 parameters:
   - name: listen_port
+    description: A port which the agent will listen for syslog messages
     type: int
     default: 514
   - name: listen_ip
+    description: A syslog ip address
```
Comment (BinaryFissionGames) [inline plugins/ubiquiti_logs.yaml:10]:
```suggestion
    description: The local IP address to listen for syslog connections on
```

File: plugins/udp_logs.yaml [hunk 13]
```diff
@@ -3,15 +3,19 @@ title: UDP
...
   - name: listen_port
+    description: A port which the agent will listen for udp messages
     type: int
     required: true
   - name: log_type
+    description: Adds the specified 'Type' as a label to each log message
     type: string
     default: udp
   - name: listen_ip
+    description: A UDP ip address
```
Comment (BinaryFissionGames) [inline plugins/udp_logs.yaml:14]:
```suggestion
    description: The local IP address to listen for UDP connections on
```

File: plugins/vcenter_logs.yaml [hunk 14]
```diff
@@ -3,22 +3,28 @@ title: VMware vCenter
 description: Log parser for VMware vCenter
 parameters:
   - name: listen_port
+    description: A port which the agent will listen for vCenter messages
     type: int
     default: 5140
   - name: listen_ip
+    description: A vCenter ip address
```
Comment (BinaryFissionGames) [inline plugins/vcenter_logs.yaml:10]:
```suggestion
    description: The local IP address to listen for vCenter connections on
```

**Review by BinaryFissionGames — COMMENTED** [review]
Took a close look, other than these few small things, I think it looks good.

**Review by BinaryFissionGames — APPROVED** [review]

**Pull request merged**